### PR TITLE
fix(StatusDateRangePicker): use StatusDatePicker for the range selection

### LINKED ISSUE #10900: [ui/wallet] make `StatusDateRangePicker` use `StatusDatePicker`
- exchange the direct input with a calendar popup (rationale below)
- extend the validation of the range (so that `from < to === true`, etc)

Needed for https://github.com/status-im/status-desktop/issues/10823

# Bug Report

TLDR; direct editing of dates is very error prone and nearly impossible due to (Q)Locale tech limitations

The current StatusDateInput is broken for anything else than English:

image

Even with the separator manually "fixed", it still returns wrong values:

image

### Additional Information

- Status desktop version: master
- Operating System: linux

### PR DESCRIPTION
**Pull request by caybro**
- exchange the direct input with a calendar popup
- extend the validation of the range (so that `from <= to === true`, can't set `to` to future, etc)
- remove direct input components (Status[Base]DateInput.qml)

Fixes #10900

### Affected areas

StatusDateRangePicker

### Screenshot of functionality (including design for comparison)

- [x] I've checked the design and this PR matches it

image

[validation] No date in future:
image

[validation] From > To:
image

### CODE REVIEW

**Review by Khushboo-dev-cpp — COMMENTED** [review]
Looks great @caybro , only thing is as per design the reset button on not on the footer but to the right of the inputs, can we update that as per design?

**Comment by caybro** [comment]
> Looks great @caybro , only thing is as per design the reset button on not on the footer but to the right of the inputs, can we update that as per design?
> 
> image

Oh, I moved it into the footer just according to the design :) Now I'm confused 

image

**Comment by caybro** [comment]
@benjthayer just double checking, where should the `Reset` button be placed?

**Comment by benjthayer** [comment]
> @benjthayer just double checking, where should the `Reset` button be placed?

Hey @caybro @Khushboo-dev-cpp , in the original designs for the MVP version of the activity filter by time, the `Reset` button was placed to the right of the date inputs. This was more an aesthetic decision - the alternatives which don't look as nice are below.

Screenshot 2023-06-05 at 12 51 14

If we feel strongly we can move the `Reset` button to the lower toolbar and perhaps go with the middle of the 2 options above - I don't feel strongly either way :)

**Comment by caybro** [comment]
> > @benjthayer just double checking, where should the `Reset` button be placed?
> 
> Hey @caybro @Khushboo-dev-cpp , in the original designs for the MVP version of the activity filter by time, the `Reset` button was placed to the right of the date inputs. This was more an aesthetic decision - the alternatives which don't look as nice are below.
> 
> Screenshot 2023-06-05 at 12 51 14
> If we feel strongly we can move the `Reset` button to the lower toolbar and perhaps go with the middle of the 2 options above - I don't feel strongly either way :)

So the conclusion, should I move them back next to the inputs? :)

**Comment by benjthayer** [comment]
> > > @benjthayer just double checking, where should the `Reset` button be placed?
> > 
> > 
> > Hey @caybro @Khushboo-dev-cpp , in the original designs for the MVP version of the activity filter by time, the `Reset` button was placed to the right of the date inputs. This was more an aesthetic decision - the alternatives which don't look as nice are below.
> > Screenshot 2023-06-05 at 12 51 14
> > If we feel strongly we can move the `Reset` button to the lower toolbar and perhaps go with the middle of the 2 options above - I don't feel strongly either way :)
> 
> So the conclusion, should I move them back next to the inputs? :)

If there are no objections, I think for the MVP version we move the `Reset` button back to be inline with the inputs. Then when we move to the post MVP version with the histogram etc, we place down in the dialog toolbar

**Comment by benjthayer** [comment]
> @Khushboo-dev-cpp @benjthayer alright, pushed an update; this is how it looks now:
> 
> image

Awesome! Thanks @caybro . One small detail on the validation state there, if we are only permitting the user to enter the date atm through the date picker, any dates ahead of today's date should be disabled - so technically the user should never see that validation. If / when we explore adding direct input of dates into the input, this validation state technically could be reached.

**Review by Khushboo-dev-cpp — APPROVED** [review]
Awesome work :)

**Review by noeliaSD — APPROVED** [review]
LGTM! Just a small comment!

File: ui/app/AppLayouts/Wallet/controls/StatusDateRangePicker.qml
[the remark this answers is not in the document]
Comment (caybro) [reply ui/app/AppLayouts/Wallet/controls/StatusDateRangePicker.qml:1]: Yeah maybe in the future; right now it's not used anywhere outside StoryBook :)

**Pull request merged**